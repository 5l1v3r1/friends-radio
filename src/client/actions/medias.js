export function updateList(name, list) {
  return {
    type: 'UPDATE_LIST',
    payload: { name, list }
  }
}
