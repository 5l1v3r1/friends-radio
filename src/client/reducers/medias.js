export default function medias(state = {}, action) {
  if (action.type === 'UPDATE_LIST') {
    return { ...state, [action.payload.name]: [...action.payload.list] }
  } else {
    return state
  }
}
