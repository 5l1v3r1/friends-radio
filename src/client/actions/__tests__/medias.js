import {updateList} from "../medias"

test("updateList", () => {
  expect(updateList("test_name", "test_list")).toMatchSnapshot()
})
