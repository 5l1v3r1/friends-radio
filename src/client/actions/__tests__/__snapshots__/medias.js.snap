// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateList 1`] = `
Object {
  "payload": Object {
    "list": "test_list",
    "name": "test_name",
  },
  "type": "UPDATE_LIST",
}
`;
